addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -124,14 +124,15 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
+        buttonExecutor = Executors.newSingleThreadScheduledExecutor();
         cancelRetry();
         connect();
     }
 
     @Override
     public void dispose() {
-        listeners.clear();
         buttonExecutor.shutdownNow();
```

Comment:
Isn't it possible to remove this instead so we do not recreate the executor, creating an executor seems like a non trivial action and dispose and re-initialize might happen quite often.